tests/test_compression.cc

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (C) 2023 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <algorithm>
+#include <random>
+
+#include <albatross/Common>
+
+#include <gtest/gtest.h>
+
+static constexpr std::size_t kMaxInputs = 1000;
+static constexpr std::size_t kNumIterations = 100;
+static constexpr std::size_t kSeed = 22;
+static constexpr int zstd_min_level = 0;
+static constexpr int zstd_max_level = 20;
+
+TEST(Compress, StringRoundtrips) {
+  std::default_random_engine generator(kSeed);
+  std::uniform_int_distribution<std::size_t> size_distribution(0, kMaxInputs);
+  std::uniform_int_distribution<int8_t> distribution('A', 'z');
+  for (std::size_t i = 0; i < kNumIterations; ++i) {
+    std::string inputs(size_distribution(generator), '\0');
+    std::generate(inputs.begin(), inputs.end(), [&generator, &distribution]() {
+      return distribution(generator);
+    });
+    const auto compressed = albatross::zstd::compress(inputs);
+    const auto outputs = albatross::zstd::decompress(compressed);
+    EXPECT_EQ(inputs, outputs);
+    std::string maybe_outputs;
+    const auto result =
+        albatross::zstd::maybe_decompress(compressed, &maybe_outputs);
+    EXPECT_TRUE(result);
+    EXPECT_EQ(inputs, maybe_outputs);
+  }
+}
+
+TEST(Compress, StringRoundtripsAlternateCompressionLevels) {
+  std::default_random_engine generator(kSeed);
+  std::uniform_int_distribution<std::size_t> size_distribution(0, kMaxInputs);
+  std::uniform_int_distribution<int8_t> distribution('A', 'z');
+  for (std::size_t i = 0; i < kNumIterations; ++i) {
+    for (int level = zstd_min_level; level <= zstd_max_level; ++level) {
+      std::string inputs(size_distribution(generator), '\0');
+      std::generate(
+          inputs.begin(), inputs.end(),
+          [&generator, &distribution]() { return distribution(generator); });
+      const auto compressed = albatross::zstd::compress(inputs, level);
+      const auto outputs = albatross::zstd::decompress(compressed);
+      EXPECT_EQ(inputs, outputs);
+      std::string maybe_outputs;
+      const auto result =
+          albatross::zstd::maybe_decompress(compressed, &maybe_outputs);
+      EXPECT_TRUE(result);
+      EXPECT_EQ(inputs, maybe_outputs);
+    }
+  }
+}
+
+TEST(Compress, DecompressEmpty) {
+  std::string inputs;
+  ASSERT_DEATH({ const auto result = albatross::zstd::decompress(inputs); },
```

Comment:
Is this something we would want to assert on? One recent issue that has come up in the AQ, the current solution is to return an empty model instead of asserting. Could this cause issues on the receiving end? @jbangelo 
https://github.com/swift-nav/orion-engine/pull/6939

Reply:
I'm not sure if we have a better way to handle failures here.  These routines get used in the `cereal` serialisation / deserialisation routines, and I'm not aware of a way to propagate deserialisation failures via the `cereal` API.

The code this is replacing also asserted on any detected decompression failure, so for this change I chose to preserve that behavior.